### PR TITLE
Rework webpack filepath to allow symlinks and remove character combin…

### LINKED ISSUE #10791: webpack doesn't build if project name contains the string "js"
In setting up a new project to test #10642 this morning I was having trouble setting up a new project.

Traced it to my project name "prjsevenfive" containing the substring "js", which violates an assumption here:

https://github.com/archesproject/arches/blob/0c89e7d117f4c639f1ce706f3dfaa82462fb1ca4/arches/webpack/webpack-utils/build-javascript-filepath-lookup.js#L20

Sneaky one! 😉

### PR DESCRIPTION
**Pull request by bferguso**
…ation collision issues (eg js)

### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Reworked the webpack file lookup logic to avoid using any special character combinations in the path that were causing issues, and updated so symlinks to outside files can be properly resolved.

### Issues Solved

Closes #10791 

### Checklist

-   I targeted one of these branches:
    - [ ] dev/7.6.x (under development): features, bugfixes not covered below
    - [x] dev/7.5.x (main support): regressions, crashing bugs, security issues, major bugs in new features
    - [ ] dev/6.2.x (extended support): major security issues, data loss issues
-   [ ] I added a changelog in arches/releases
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [ ] Unit tests pass locally with my changes
-   [ ] I added tests that prove my fix is effective or that my feature works
-   [ ] My test fails on the target branch

#### Accessibility Checklist

[Developer Guide](https://arches.readthedocs.io/en/stable/developing/advanced/accessibility/)

| Topic            | Changed | Retested |
| ---------------- | ------- | -------- |
| Color contrast   |         |          |
| Form fields      |         |          |
| Headings         |         |          |
| Links            |         |          |
| Keyboard         |         |          |
| Responsive Design|         |          |
| HTML validation  |         |          |
| Screen reader    |         |          |

#### Ticket Background
*   Sponsored by: Self Funded
*   Found by: @bferguso
*   Tested by: @bferguso
*   Designed by: @bferguso

### Further comments
Have not done any ancillary work to complete this PR. Submitting for feedback.
Collapsed JS, Template and Image file lookup functions into a single function as most of the logic was the same. Didn't make changes to the vue lookup function as I don't have any vue components to test against.

### CODE REVIEW

File: arches/install/arches-templates/project_name/webpack/webpack-utils/build-filepath-lookup.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const Path = require('path');
+const fs = require('fs');
+
+function buildFilePathLookup(path, staticUrlPrefix) {
+    if (!fs.existsSync(path)) {
+        return;
+    }
+
+    let prefix = path.match(/[^\/]+$/)
```
Comment (chrabyrd) [inline arches/install/arches-templates/project_name/webpack/webpack-utils/build-filepath-lookup.js:9]: nit throughout -- default to `const` and use `let` only if expecting reassignment

Comment (chrabyrd) [inline arches/install/arches-templates/project_name/webpack/webpack-utils/build-filepath-lookup.js:9]: nit L9/l10, needs terminating semicolon

File: arches/install/arches-templates/project_name/webpack/webpack.common.js [hunk 2]
```diff
@@ -349,6 +356,8 @@ module.exports = () => {
                                     removeComments: false,
                                 },
                                 preprocessor: async (content, loaderContext) => {
+                                    // console.log(`content: ${content}`);
```
Comment (chrabyrd) [inline arches/install/arches-templates/project_name/webpack/webpack.common.js:359]: nit: remove comments

File: arches/install/arches-templates/project_name/webpack/webpack.common.js [hunk 3]
```diff
@@ -33,35 +31,44 @@ module.exports = () => {
...
             }, {});
 
             // END create entry point configurations
             // BEGIN create JavaScript filepath lookups
 
             const archesCoreJavascriptRelativeFilepathToAbsoluteFilepathLookup = Object.entries(archesCoreEntryPointConfiguration).reduce((acc, [path, config]) => {
-                acc[path + '$'] = Path.resolve(__dirname, path, config['import']);
+                if ((typeof config) == "object")
+                    acc[path + '$'] = Path.resolve(__dirname, path, config['import']);
+                else
```
Comment (chrabyrd) [inline arches/install/arches-templates/project_name/webpack/webpack.common.js:51]: how were you able to hit these else conditionals?

**Review by chrabyrd — COMMENTED** [review]
Overall this is looking great 👍 . Per our slack convo you're not seeking approval/merge yet but first brush and test run look good, and this creates an identical bundle to `dev/7.5.x`

File: arches/install/arches-templates/project_name/webpack/webpack-utils/build-filepath-lookup.js (hunk 1 above)
Comment (bferguso) [reply arches/install/arches-templates/project_name/webpack/webpack-utils/build-filepath-lookup.js]: Resolved.

File: arches/install/arches-templates/project_name/webpack/webpack.common.js (hunk 2 above)
Comment (bferguso) [reply arches/install/arches-templates/project_name/webpack/webpack.common.js]: Removed.

File: arches/install/arches-templates/project_name/webpack/webpack.common.js (hunk 3 above)
Comment (bferguso) [reply arches/install/arches-templates/project_name/webpack/webpack.common.js:51]: This happens if there's a non-js file in the `js/` tree. The object type that comes back is dependent on the extension (js files have an object config with name/import keys, the rest are straight strings). I was actually wondering if we should make the value returned from the file lookup builder consistent for all asset types, which would eliminate having to do this here, and make things more obvious from the caller.

Comment (chrabyrd) [reply arches/install/arches-templates/project_name/webpack/webpack.common.js:51]: Yeah I'd definitely say it's would make things neater and more readable, if it doesn't look too complex to pull off 👍

Comment (chrabyrd) [reply arches/install/arches-templates/project_name/webpack/webpack.common.js:51]: When implementing this in 7.6 I did not make these changes and everything still built correctly. Happy to resolve the differences between branches either way and just wanted to flag this 🏴

**Comment by chrabyrd** [comment]
Overall looks good! After our slack conversation I was able to incorporate this in 7.6 with [this PR](https://github.com/archesproject/arches/pull/11025). I'll add you as a reviewer 👍 

There are some minor discrepancies between our implementations that should be reconciled -- happy to chat about them any time

**Review by chrabyrd — APPROVED** [review]
Just made some very minor adjustments as one last nit, lgtm! 👍

**Pull request merged**